dm/syncer/dml.go

```diff
@@ -94,6 +98,10 @@ func (s *Syncer) genAndFilterInsertDMLs(tctx *tcontext.Context, param *genDMLPar
 	}
 	downstreamIndexColumns := downstreamTableInfo.AbsoluteUKIndexInfo
 
+	if len(extendValues) > 0 {
+		originalDataSeq = extendData
```

Comment:
in line 114, `extractValueFromData` has assumed each row in `originalDataSeq` has same length with `ti.Columns`. But now the assumption is broken.

by a brief inspecting, that branch will be triggered when upstream table has a generated column. Also I think it's good to add an integration test to make sure this behaviour.

Reply:
[`extendData != nil`](https://github.com/pingcap/ticdc/blob/43456cc356332066c4a65b0446b9507b381cfe8a/dm/syncer/dml.go#L102) and [`len(columns) != len(ti.Columns)`](https://github.com/pingcap/ticdc/blob/43456cc356332066c4a65b0446b9507b381cfe8a/dm/syncer/dml.go#L114)

You mean that we need to test the behavior of [extractValueFromData](https://github.com/pingcap/ticdc/blob/43456cc356332066c4a65b0446b9507b381cfe8a/dm/syncer/dml.go#L114) when these conditions are met?

Reply:
@lance6716 @Ehco1996 One more unresolved conversation, please give me some guidance.

Reply:
as @lance6716 advice, i think just add a it-test to proof of correctness when met this condition 

* upstream table has a generated column
* downstream has extend colum

Reply:
Got it, let me see it.

Reply:
Test generate and extend columns. https://github.com/pingcap/ticdc/pull/3262/commits/8036fc41bd8c85f390b9ecb10de34af6a324e3d4